torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1871,6 +1871,22 @@ def uv_loss(usv):
 
     return out
 
+
+def sample_inputs_permute(op_info, device, dtype, requires_grad, **kwargs):
+    make_arg = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+
+    cases = [((1, 2, 3, 4), (0, 2, 3, 1)),
+             ((1, 2, 3, 4), (0, -2, -1, 1)),
+             ((), ()),
+             ((1, 2, 3, 4), (2, 1, 3, 0))]
+
+    def generator():
+        for shape, args in cases:
+            yield SampleInput(make_arg(shape), args=(args,))
+
+    return list(generator())
```

Comment:
Wouldn't it be better to use list comprehension if you're going to return as a list type anyway?

Reply:
AFAIK plan is to move to generators eventually. It would be easier to migrate with this approach (just drop the list call)